src/io/GeantGeometryMap.cc

```diff
@@ -0,0 +1,109 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantGeometryMap.cc
+//---------------------------------------------------------------------------//
+#include "GeantGeometryMap.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct/destruct with defaults
+ */
+GeantGeometryMap::GeantGeometryMap()  = default;
+GeantGeometryMap::~GeantGeometryMap() = default;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the mat_id for a given vol_id
+ */
+GeantGeometryMap::mat_id GeantGeometryMap::get_matid(vol_id& volume_id)
+{
+    auto iter = volid_to_matid_.find(volume_id);
+    REQUIRE(iter != volid_to_matid_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantVolume associated with the vol_id
+ */
+GeantVolume GeantGeometryMap::get_volume(vol_id& volume_id)
+{
+    auto iter = volid_to_volume_.find(volume_id);
+    REQUIRE(iter != volid_to_volume_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantMaterial associated with the mat_id
+ */
+GeantMaterial GeantGeometryMap::get_material(mat_id& material_id)
+{
+    auto iter = matid_to_material_.find(material_id);
+    REQUIRE(iter != matid_to_material_.end());
+    return iter->second;
+}
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantElement associated with the elem_id
+ */
+GeantElement GeantGeometryMap::get_element(elem_id& element_id)
+{
+    auto iter = elemid_to_element_.find(element_id);
+    REQUIRE(iter != elemid_to_element_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return a copy of private member volid_to_matid_
+ */
+std::map<GeantGeometryMap::vol_id, GeantGeometryMap::mat_id>
+GeantGeometryMap::volid_to_matid_map()
+{
+    return volid_to_matid_;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Add pair <mat_id, GeantMaterial> to the matid_to_material_ map
+ */
+void GeantGeometryMap::add_material(mat_id id, GeantMaterial material)
+{
+    matid_to_material_.insert(std::pair<mat_id, GeantMaterial>(id, material));
```

Comment:
With C++11 you can just write `insert({id, material})`: much easier. (Even in C++98 you could write `make_pair(id, material)`.) If you think that no one should be inserting duplicates -- which you probably should -- write
```suggestion
    auto result = matid_to_material_.insert(std::pair<mat_id, GeantMaterial>(id, material));
    CHECK(result.second);
```
since `result` is a `pair<iter, bool>` where the second element is "true" if the insertion was successful (i.e. the ID is unique).

Finally, make your `GeantMaterial` value in the argument a const reference to avoid a needless copy.

Reply:
Done.